### PR TITLE
fix edge case in btcclient.GetBlockHeightByTimestamp

### PR DESCRIPTION
**Pull request by bap2pecs**
## Summary

when we were testing fast finality in our demo dApp, we saw error

```
Received request to query if block is finalized: {
  blockHash: '0xdfb4262e3086cd2bcc9dc7d26d29f513ade4cb4432d3b568134d9345da15efb1',
  blockHeight: 3510,
  blockTimestamp: 1726412271
}
Error: 2 UNKNOWN: BTC staking is not activated for the consumer chain
```

this is b/c when we query a tx's finalized status immediately after it's confirmed, the timestamp will likely be higher than the latest BTC timestamp. so `GetBlockHeightByTimestamp` will return 0. so in finalitygadget/finalitygadget.go, `QueryIsBlockBabylonFinalized` will throw:

```
func (fg *FinalityGadget) QueryIsBlockBabylonFinalized(block *types.Block) (bool, error) {
        ...        
        // convert the L2 timestamp to BTC height
	btcblockHeight, err := fg.btcClient.GetBlockHeightByTimestamp(block.BlockTimestamp)
	if err != nil {
		return false, err
	}

	// check whether the btc staking is actived
	earliestDelHeight, err := fg.bbnClient.QueryEarliestActiveDelBtcHeight(allFpPks)
	if err != nil {
		return false, err
	}
	if btcblockHeight < earliestDelHeight { // throws here b/c `btcblockHeight` is 0
		return false, types.ErrBtcStakingNotActivated
	}
        ...
}
```

so instead of returning 0, we return `math.MaxUint64` to indicate that

---

this PR also does a few things
- fix existing lint issues
- fix failed tests in `main`
- remove `babylon-private` dependency
- enable CI
- cors: allow access from any origin

## Test Plan

```
make lint
make test
```

TODOs:
- build the binary and test e2e on our demonet

### CODE REVIEW

File: types/transaction.go [hunk 1]
```diff
@@ -2,10 +2,10 @@ package types
 
 type TransactionInfo struct {
 	TxHash           string         `json:"txHash"`
-	BlockTimestamp   uint64         `json:"blockTimestamp"`
-	BlockHeight      uint64         `json:"blockHeight"`
 	BlockHash        string         `json:"blockHash"`
 	Status           FinalityStatus `json:"status"`
+	BlockTimestamp   uint64         `json:"blockTimestamp"`
```
Comment (bap2pecs) [inline types/transaction.go:7]: this fixes lint

```
$ make lint                                                        [11:14:45]
golangci-lint run
types/transaction.go:3:22: fieldalignment: struct with 56 pointer bytes could be 40 (govet)
type TransactionInfo struct {
                     ^
make: *** [lint] Error 1
```

**Comment by bap2pecs** [comment]
seems `make test` is broken on `main`. i will fix it all together in this PR and also enable CI

jfyi @lesterli @parketh @SebastianElvis

**Comment by bap2pecs** [comment]
@SebastianElvis any idea why CI is not enabled in this repo?

**Review by SebastianElvis — APPROVED** [review]

**Comment by SebastianElvis** [comment]
> any idea why CI is not enabled in this repo?

Github action does not seem to be enabled for this repo. cc @babylon-devops any ideas?

**Pull request merged**